google/cloud/sql/connector/InstanceConnectionManager.py

```diff
@@ -38,12 +37,32 @@
 
 
 class InstanceMetadata:
-    ip_addresses: Dict[str, str]
-    ssl_context: OpenSSL.SSL.Context
+    ip_address: str
+    ca_fileobject: NamedTemporaryFile
+    cert_fileobject: NamedTemporaryFile
+    key_fileobject: NamedTemporaryFile
 
-    def __init__(self, ssl_context, ip_addresses):
-        self.ssl_context = ssl_context
-        self.ip_addresses = ip_addresses
+    def __init__(
+        self,
+        ephemeral_cert: str,
+        ip_address: str,
+        private_key: str,
+        server_ca_cert: str,
+    ):
+        self.ip_address = ip_address
+
+        self.ca_fileobject = NamedTemporaryFile(suffix=".pem")
+        self.cert_fileobject = NamedTemporaryFile(suffix=".pem")
+        self.key_fileobject = NamedTemporaryFile(suffix=".pem")
+
+        write_to_file(server_ca_cert, ephemeral_cert, private_key)
+
+        self.ca_fileobject.write(server_ca_cert.encode())
+        self.ca_fileobject.seek(0)
```

Comment:
Why do you call seek? This section would be a good place to put a comment explaining what/why you are doing this (since it's not obvious from reading the code)